### PR TITLE
Support for VCN Native Pod Networking

### LINKED ISSUE #539: Add support for native pod networking
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for native pod networking when it's available in the provider: https://blogs.oracle.com/cloud-infrastructure/post/announcing-vcn-native-pod-networking-for-kubernetes-in-oci

### Potential Terraform Configuration

```hcl
# Copy-paste any Terraform configurations for how the requested feature may be used. 
```

### PR DESCRIPTION
**Pull request by hyder**
Closes #539

### CODE REVIEW

**Review by snafuz — APPROVED** [review]

**Comment by hyder** [comment]
@Djelibeybi can you please review this as well. It was opened before #546 which is already merged now.

File: modules/network/locals.tf [hunk 1]
```diff
@@ -91,15 +93,16 @@ locals {
       destination      = local.osn,
       destination_type = "SERVICE_CIDR_BLOCK",
       protocol         = local.tcp_protocol,
-      port             = 443,
-      stateless        = false
+      # port           = 443
```
Comment (Djelibeybi) [inline modules/network/locals.tf:96]: This commented out line is a duplicate which is two reasons to remove it.

File: modules/network/locals.tf [hunk 2]
```diff
@@ -149,17 +152,19 @@ locals {
...
   workers_egress = [
+    # flannel only moved to a rule on its own
+    # {
+    #   description      = "Allow egress for all traffic to allow pods to communicate between each other on different worker nodes on the worker subnet",
+    #   destination      = local.workers_subnet,
+    #   destination_type = "CIDR_BLOCK",
+    #   protocol         = local.all_protocols,
+    #   port             = -1,
+    #   stateless        = false
+    # },
```
Comment (Djelibeybi) [inline modules/network/locals.tf:163]: Why did you add a block of commented out code?

File: modules/network/locals.tf [hunk 3]
```diff
@@ -149,17 +152,19 @@ locals {
...
-      destination_type = "CIDR_BLOCK",
-      protocol         = local.all_protocols,
-      port             = -1,
-      stateless        = false
-    },
-    {
-      description      = "Allow ICMP traffic for path discovery",
-      destination      = local.workers_subnet
+      description = "Allow ICMP traffic for path discovery",
+      # destination      = local.workers_subnet
```
Comment (Djelibeybi) [inline modules/network/locals.tf:166]: Please remove this duplicated and commented out line.

File: variables.tf [hunk 4]
```diff
@@ -463,6 +464,12 @@ variable "allow_worker_internet_access" {
   type        = bool
 }
 
+variable "allow_pod_internet_access" {
+  default     = true
+  description = "Allow pods to egress to internet. Required if the pods are invoking Internet services."
```
Comment (Djelibeybi) [inline variables.tf:469]: I would perhaps rephrase this as "Allow pod egress to the internet." 

Also, we should check the style guide to see if Oracle refers to "the Internet" or "the internet" and use that form throughout.

**Review by Djelibeybi — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 4 above)
Comment (hyder) [reply variables.tf:469]: "internet" wins on the style guide. Also, for future reference https://en.wikipedia.org/wiki/Capitalization_of_Internet

**Review by Djelibeybi — APPROVED** [review]
